File: 백준/Silver/25206. 너의 평점은/너의 평점은.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;
public class Main {
    static double score(String grade) {
        switch (grade) {
            case "A+": return 4.5;
            case "A0": return 4.0;
            case "B+": return 3.5;
            case "B0": return 3.0;
            case "C+": return 2.5;
            case "C0": return 2.0;
            case "D+": return 1.5;
            case "D0": return 1.0;
            case "F": return 0.0;
            default: return -1.0;
        }
    }
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        double sum = 0.0;
        double total = 0.0;
        for (int i = 0; i < 20; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            st.nextToken();
            double credit = Double.parseDouble(st.nextToken());
            String grade = st.nextToken();
            if (grade.equals("P")) continue;
            double s = score(grade);
            sum += credit * s;
            total += credit;
        }
        System.out.print(sum / total);
    }
}

File: 백준/Silver/25206. 너의 평점은/너의 평점은.cc
#include <iostream>
#include <string>
using namespace std;
double score(string grade) {
    if (grade == "A+") return 4.5;
    if (grade == "A0") return 4.0;
    if (grade == "B+") return 3.5;
    if (grade == "B0") return 3.0;
    if (grade == "C+") return 2.5;
    if (grade == "C0") return 2.0;
    if (grade == "D+") return 1.5;
    if (grade == "D0") return 1.0;
    if (grade == "F") return 0.0;
    return -1.0;
}
int main() {
    string subject, grade;
    double credit, sum = 0.0, total = 0.0;
    for (int i = 0; i < 20; i++) {
        cin >> subject >> credit >> grade;
        if (grade == "P") continue;
        sum += credit * score(grade);
        total += credit;
    }
    cout << sum / total;
    return 0;
}